src/memory/memory.ts

```diff
@@ -50,7 +50,7 @@ export class Memory {
 }
 
 export class MemorySegmentManager {
-  segmentSizes: Record<Uint, Uint>;
+  segmentSizes: Record<number, Uint>;
```

Comment:
Any reason to change this? I don't think the segments index can be negative, so having Uint makes sense no?

Reply:
I'm not allowed to have a non-native type as key, so it has to be number or string,
since former Uint was number wrapped with addons, it was fine, now that its a bigint , i cannot